app/services/repository_service.rb

```diff
@@ -37,6 +37,19 @@ def create_repository!(product, url, attributes, custom: false)
     repository
   end
 
+  def update_repository!(repo_data)
+    uri = URI(repo_data[:url])
+    auth_token = uri.query
+
+    Repository.find_by!(scc_id: repo_data[:id]).update!(
+      auth_token: auth_token,
+      enabled: repo_data[:enabled],
+      autorefresh: repo_data[:autorefresh],
+      external_url: "#{uri.scheme}://#{uri.host}#{uri.path}",
```

Comment:
We need make sure a trailing `/` is in the url otherwise we might get into trouble in other places since RMT expects this type of behavior

```suggestion
      external_url: "#{uri.scheme}://#{uri.host}#{uri.path}/",
```

Reply:
All repos coming from SCC already have a trailing slash, adding another one would be wrong. 
Instead, I've added a before_save callback to `Repository`, to assure the urls always have a trailing slash.

Reply:
I reverted the before_save callback, because we have `spec/lib/rmt/cli/repos_custom_spec.rb:147`, which expects that there is no trailing slash in case of a custom repo with a query string.